### PR TITLE
Make DimacsCnf constructors copy deeply

### PR DESCRIPTION
**Pull request by Elscrux**

### CODE REVIEW

File: src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/DimacsCnf.java [hunk 1]
```diff
@@ -38,8 +38,15 @@ public DimacsCnf(ArrayList<ArrayList<Variable>> orClauses) {
...
-    this.variables = List.copyOf(variables);
-    this.orClauses = new ArrayList<>(orClauses);
+    this.variables = variables.stream()
+            .map(Variable::new)
+            .collect(Collectors.toCollection(ArrayList::new));
+
+    this.orClauses = orClauses.stream()
+            .map(list -> list.stream()
+                    .map(Variable::new)
+                    .collect(Collectors.toCollection(ArrayList::new)))
```
Comment (schweikart) [inline src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/DimacsCnf.java:48]: please apply the DRY principle here and extract the List<Variable> copying to its own (private) function

File: src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/Variable.java [hunk 2]
```diff
@@ -3,6 +3,10 @@
 import java.util.Objects;
 
 public record Variable(int number, String name, boolean isNegated) {
+  public Variable(Variable other) {
+    this(other.number, other.name, other.isNegated);
+  }
+
```
Comment (schweikart) [inline src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/Variable.java:9]: Please document new public methods so we don't pile up new documentation debt

**Review by schweikart — CHANGES_REQUESTED** [review]

**Comment by Elscrux** [comment]
@schweikart I have included your suggestions

**Review by schweikart — APPROVED** [review]
LGTM! :)

**Pull request merged**